test/set.js

```diff
@@ -22,6 +22,22 @@ describe('set', function() {
     assert.deepEqual(b, {w: 1, x: {a: 5, b: 2, c: 3}, z: 4});
   });
 
+  it('handles empty object', function() {
+    var a = {w: 1, x: {a: 1, b: 2, c: 3}, z: 4};
+    var b = R.set({}, a);
+    assert.deepEqual(b, a);
+  });
+
+  it('handles nested empty object', function() {
+    var a = {w: 1, x: {a: 1, b: 2, c: 3}, z: 4};
+    var b = R.set({w: 0, x: {}, z: 3}, a);
+    assert.deepEqual(b, {w: 0, x: {a: 1, b: 2, c: 3}, z: 3});
+  });
+
```

Comment:
@kedashoe: I agree that something seems to be going awry here with the naming.  `assoc` has a useful cognate in other languages, so seemed like a good choice, especially because at the time `set` sounded to me very much like it mutated the object.  I'm really not clear enough on the API here, nor on the name.  I will have to go back over the recent history to determine why we deprecated `merge`, as it doesn't seem to be covered by anything else.  But I think mostly I'm just confused at the moment.